### PR TITLE
Temporary fix for failure to resolve 'sun.management.Util' when using GraalVM CE 21.3.0 Java 17  

### LINKED ISSUE #21372: Integration test jgit fails to compile natively with GraalVM CE 21.3 Java 17
### Describe the bug

Native compilation of integration test `jgit` fails with GraalVM CE 21.3 Java17 (`quay.io/quarkus/ubi-quarkus-native-image:21.3-java17`)

See https://github.com/graalvm/mandrel/runs/4170345504?check_suite_focus=true

Note that this is not a Quarkus issue. The underlying issue is only reproducible on Java 17 and has been fixed upstream in https://github.com/oracle/graal/issues/3875 (and will probably land in the January releases)

### Expected behavior

jgit test should build and run.

### Actual behavior

Test fails to build with:
```
Error: Unsupported features in 8 methods
Detailed message:
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: sun.management.Util. To diagnose the issue you can use the --allow-incomplete-classpath option. The missing type is then reported at run time when it is accessed the first time.
Trace: 
	at parsing com.oracle.svm.core.genscavenge.CompleteGarbageCollectorMXBean.getObjectName(CompleteGarbageCollectorMXBean.java:81)
Call path from entry point to com.oracle.svm.core.genscavenge.CompleteGarbageCollectorMXBean.getObjectName(): 
	at com.oracle.svm.core.genscavenge.CompleteGarbageCollectorMXBean.getObjectName(CompleteGarbageCollectorMXBean.java:81)
	at com.oracle.svm.core.jdk.management.ManagementSupport.addMXBean(ManagementSupport.java:267)
	at com.oracle.svm.core.jdk.management.ManagementSupport.getPlatformMBeanServer(ManagementSupport.java:259)
	at com.oracle.svm.core.jdk.management.Target_java_lang_management_ManagementFactory.getPlatformMBeanServer(Target_java_lang_management_ManagementFactory.java:45)
	at org.eclipse.jgit.util.Monitoring.registerMBean(Monitoring.java:69)
	at org.eclipse.jgit.internal.storage.file.WindowCache.publishMBeanIfNeeded(WindowCache.java:485)
	at org.eclipse.jgit.internal.storage.file.WindowCache.get(WindowCache.java:386)
	at org.eclipse.jgit.internal.storage.file.WindowCursor.pin(WindowCursor.java:327)
	at org.eclipse.jgit.internal.storage.file.WindowCursor.copy(WindowCursor.java:226)
	at org.eclipse.jgit.internal.storage.file.PackInputStream.read(PackInputStream.java:37)
	at com.jcraft.jsch.ChannelSession.run(ChannelSession.java:245)
	at java.lang.Thread.run(Thread.java:833)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:596)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
	at com.oracle.svm.core.code.IsolateEnterStub.PosixJavaThreads_pthreadStartRoutine_e1f4a8c0039f8337338252cd8734f63a79b5e3df(generated:0)
...
```

### How to Reproduce?

```
./mvnw -Dnative -Dnative.surefire.skip -pl integration-tests/jgit package -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:21.3-java17
```

### Output of `uname -a` or `ver`

5.14.15-200.fc34.x86_64 #1 SMP Wed Oct 27 15:53:30 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

OpenJDK 64-Bit Server VM Temurin-11.0.13+8 (build 11.0.13+8, mixed mode)

### GraalVM version (if different from Java)

OpenJDK 64-Bit Server VM GraalVM CE 21.3.0 (build 17.0.1+12-jvmci-21.3-b05, mixed mode, sharing)

### Quarkus version or git rev

132d9ea09b795d6027d557f46627092cfb29ce29

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)

### PR DESCRIPTION
**Pull request by zakkak**
This PR:
1. updates the GraalVM.Version implementation to use `org.graalvm.home.Version` as its backend allowing us to perform comparisons beyond the major.minor part of the version without reinventing the wheel.
2. extends GraalVM.Version to include the base JDK version (e.g. 17) of GraalVM
3. Uses the above two to apply work around for issue #21372 only when using GraalVM 21.3.0 Java 17 

This work-around is only necessary for 21.3.0 as it is already fixed upstream with https://github.com/oracle/graal/pull/3949 which is expected to land in 22.0 and 21.3.1 ([pending confirmation](https://github.com/oracle/graal/pull/3949#issuecomment-967284443)) in January.

Closes: #21372

### CODE REVIEW

**Comment by indalaterre** [comment]
Is it possible to create a patch of 2.4 for this issue?

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java [hunk 1]
```diff
@@ -8,24 +8,28 @@
...
-        static final Version UNVERSIONED = new Version("Undefined", -1, -1, Distribution.ORACLE);
-        static final Version VERSION_21_2 = new Version("GraalVM 21.2", 21, 2, Distribution.ORACLE);
-        static final Version VERSION_21_3 = new Version("GraalVM 21.3", 21, 3, Distribution.ORACLE);
+        static final Version UNVERSIONED = new Version("Undefined", "snapshot", Distribution.ORACLE);
+        static final Version VERSION_21_2 = new Version("GraalVM 21.2", "21.2", Distribution.ORACLE);
+        static final Version VERSION_21_3_0 = new Version("GraalVM 21.3.0", "21.3.0", Distribution.ORACLE);
 
         static final Version MINIMUM = VERSION_21_2;
-        static final Version CURRENT = VERSION_21_3;
+        static final Version CURRENT = VERSION_21_3_0;
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java:18]: I'm not sure we want that? IIRC, we use `CURRENT` in some error messages and I'm pretty sure we prefer advertising the minor and not the micro. Or am I mistaken?

If I were you, I would have kept `VERSION_21_3`, kept it as the current version and introduce `VERSION_21_3_0` as a brand new constant.

**Review by gsmet — CHANGES_REQUESTED** [review]

**Comment by gsmet** [comment]
@indalaterre we might release another 2.4 this week but we'll see if this particular patch makes it.

In the meantime, you should be able to work around the issue by adding this line to your `application.properties`:
```
quarkus.native.additional-build-args=-J--add-exports=java.management/sun.management=ALL-UNNAMED
```

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java (hunk 1 above)
Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java]: > I'm not sure we want that? IIRC, we use `CURRENT` in some error messages and I'm pretty sure we prefer advertising the minor and not the micro. Or am I mistaken?

You are right, thanks!

**Review by gsmet — APPROVED** [review]
Cool, thanks!

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/DisableIfBuiltWithGraalVMOlderThan.java [hunk 2]
```diff
@@ -46,19 +40,14 @@ public int getMinor() {
          * @return {@code -1} if this version is older than the version represented by the supplied major and minor parts,
```
Comment (gastaldi) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/DisableIfBuiltWithGraalVMOlderThan.java:40]: Probably this javadoc needs to be updated too

**Comment by gsmet** [comment]
Unfortunately, we have several CI issues. I'll rebase when they are fixed.

**Review by gsmet — APPROVED** [review]

**Pull request merged**